### ISSUE TITLE
Origin integration offline, doesn't sync playtimes

### ISSUE THREAD

**Issue by nicksundermeyer** [FriendsOfGalaxy/galaxy-integration-origin#16]
**Describe the bug**
The Origin integration is installed and I am logged into my account, and it can show and launch games. However, in the integration settings it says that it is offline, and it doesn't sync playtime or recently played games.

**Expected behavior**
Playtime and recently played should sync, and the plugin should not show as offline.

**Attached logs**
[plugin-origin-7f53219b-4e2b-4591-9f4f-dfc5f4ba9eb0.log](https://github.com/FriendsOfGalaxy/galaxy-integration-origin/files/4264006/plugin-origin-7f53219b-4e2b-4591-9f4f-dfc5f4ba9eb0.log)

**Comment by PastorGL** [comment]
I observe the same. Relevant log portion:

```
2020-03-02 23:25:08,139 - galaxy.api.plugin - ERROR - Error while running plugin
Traceback (most recent call last):
  File "D:\obj\Windows-Release\37win32_Release\msi_python\zip_win32\windows_events.py", line 435, in finish_recv
OSError: [WinError 64] The specified network name is no longer available

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\x\AppData\Local\GOG.com\Galaxy\plugins\installed\origin_7f53219b-4e2b-4591-9f4f-dfc5f4ba9eb0\galaxy\api\plugin.py", line 1045, in create_and_run_plugin
    asyncio.run(coroutine())
  File "D:\obj\Windows-Release\37win32_Release\msi_python\zip_win32\runners.py", line 43, in run
  File "D:\obj\Windows-Release\37win32_Release\msi_python\zip_win32\base_events.py", line 584, in run_until_complete
  File "C:\Users\x\AppData\Local\GOG.com\Galaxy\plugins\installed\origin_7f53219b-4e2b-4591-9f4f-dfc5f4ba9eb0\galaxy\api\plugin.py", line 1038, in coroutine
    await writer.wait_closed()
  File "D:\obj\Windows-Release\37win32_Release\msi_python\zip_win32\streams.py", line 323, in wait_closed
  File "C:\Users\x\AppData\Local\GOG.com\Galaxy\plugins\installed\origin_7f53219b-4e2b-4591-9f4f-dfc5f4ba9eb0\galaxy\api\jsonrpc.py", line 167, in run
    data = await self._reader.readline()
  File "C:\Users\x\AppData\Local\GOG.com\Galaxy\plugins\installed\origin_7f53219b-4e2b-4591-9f4f-dfc5f4ba9eb0\galaxy\reader.py", line 15, in readline
    chunk = await self._reader.read(1024*1024)
  File "D:\obj\Windows-Release\37win32_Release\msi_python\zip_win32\streams.py", line 640, in read
  File "D:\obj\Windows-Release\37win32_Release\msi_python\zip_win32\streams.py", line 473, in _wait_for_data
  File "D:\obj\Windows-Release\37win32_Release\msi_python\zip_win32\proactor_events.py", line 255, in _loop_reading
  File "D:\obj\Windows-Release\37win32_Release\msi_python\zip_win32\windows_events.py", line 744, in _poll
  File "D:\obj\Windows-Release\37win32_Release\msi_python\zip_win32\windows_events.py", line 439, in finish_recv
ConnectionResetError: [WinError 64] The specified network name is no longer available
2020-03-02 23:26:22,199 - root - DEBUG - Using proactor: IocpProactor
2020-03-02 23:26:22,202 - galaxy.api.plugin - INFO - Using local address: 127.0.0.1:52114
2020-03-02 23:26:22,202 - galaxy.api.plugin - INFO - Creating plugin for platform origin, version 0.33
2020-03-02 23:26:22,283 - galaxy.api.jsonrpc - DEBUG - Received 66 bytes of data
2020-03-02 23:26:22,284 - galaxy.api.jsonrpc - INFO - Handling request: id=0, method=get_capabilities, params={}
2020-03-02 23:26:22,284 - root - DEBUG - Sending data: {"jsonrpc": "2.0", "id": "0", "result": {"platform_name": "origin", "features": ["ImportFriends", "ImportGameTime", "ImportOwnedGames", "InstallGame", "ShutdownPlatformClient", "ImportGameLibrarySettings", "ImportInstalledGames", "UninstallGame", "ImportAchievements", "LaunchGame"], "token": "****"}}
2020-03-02 23:26:22,401 - galaxy.api.jsonrpc - DEBUG - Received 300730 bytes of data
2020-03-02 23:26:22,408 - galaxy.api.jsonrpc - INFO - Handling request: id=1, method=initialize_cache, params={'data': '****'}
2020-03-02 23:26:22,412 - galaxy.task_manager - DEBUG - Task manager plugin internal: creating task 0 (tick)
2020-03-02 23:26:22,413 - root - DEBUG - Sending data: {"jsonrpc": "2.0", "id": "1", "result": null}
2020-03-02 23:26:22,435 - galaxy.api.jsonrpc - DEBUG - Received 54 bytes of data
2020-03-02 23:26:22,435 - galaxy.api.jsonrpc - INFO - Handling request: id=2, method=ping, params={}
2020-03-02 23:26:22,436 - root - DEBUG - Sending data: {"jsonrpc": "2.0", "id": "2", "result": null}
```

**Comment by FriendsOfGalaxy** [comment]
Hi, thanks for the report. 

The OP problem is:
`ValueError: time data '2019-02-03T06:35:55Z' does not match format '%Y-%m-%dT%H:%M:%S.%fZ'`

@PastorGL issue can be probably fixed by rebuilding with new version of aiohttp. https://github.com/aio-libs/aiohttp/issues/3648
(API requires aiohttp>=3.5.4), so it should be build fixed automatically with the new release.

**Comment by FriendsOfGalaxy** [comment]
This should be fixed in 0.34. Closing.

@PastorGL "ConnectionResetError" occurs when socket is closed before plugin shutdown is done.  I can repro this when I kills Galaxy forcibly. But this issue does not breaks anything, and may be suppress.

**Comment by nicksundermeyer** [comment]
This appears to have fixed it, thank you!
